### DISCUSSION TITLE
How to save visibly hidden data?

### DISCUSSION THREAD

**Discussion by moskoweb** [filamentphp/filament#5482] [Help]
I have a form that has some elements that appear only when activating the feature through a Toggle, but I realized that when you activate Toggle, enter a value in the MarkdownEditor field that appears, saves, and then in editing, you disable Toggle, the TextInput value is lost.

```php
Components\Toggle::make('welcome.status')
    ->inline(false)
    ->reactive(),

Components\MarkdownEditor::make('welcome.content')
    ->visible(fn (\Closure $get) => $get('welcome.status') === true),
```

I also tested using the "hidden", and it has the same behavior, after it disables the Status, it deletes the content that was in the Content, and the user can disable and reactivate without losing any value.

**Comment by micraux** [comment]
There is a feature that allows you to have different behaviors depending on the context (`CREATE` or `EDIT`). I'm sorry, I cant' find the relevant documentation: I know it is documented as I saw it previously but I'm unable to find it right now!

1. First idea
You can use this feature to disable the `->reactive()` in your Toggle.

2. Second idea
You may also be interested by the `afterStateUpdated()` method: [https://filamentphp.com/docs/3.x/forms/advanced#field-updates](https://filamentphp.com/docs/3.x/forms/advanced#field-updates). The idea is when you are in editing mode, you can use the `afterStateUpdated()` method on your Toggle  in order to "not save" the MarkdownEditor.

I don't know if this can help you...
